docs/upload_specifications.ADR.md

```diff
@@ -0,0 +1,170 @@
+# Upload Specifications Architecture Design Record
+
+This document specifies the design for handling upload specifications in the Staging Service (StS).
+An upload specification is an Excel, CSV, or TSV file that contains instructions for how
+to upload one or more files in the staging area to KBase as KBase data types.
+
+## Resources
+
+* [The original strategy document for this approach](https://docs.google.com/document/d/1ocmZVBlTzAh_cdZaWGRwIbAuH-mcPRZFdhcwRhAfzxM/edit)
+  * Readable for everyone that has access to the KBase Google Docs folder
+  * The document contains short descriptions of future projects not included in this work,
+    such as generating template files on the fly and rich templates.
+
+## Front end changes
+
+The design introduces a new StS data type, `upload_specification`. The FE's current 
+behavior is to display any data types in the file dropdown, but ignore files for which
+the selected data type is unknown, a bug. The FE will be updated to ignore unknown
+data types returned from the StS, allowing for phased, non-lockstep upgrades. This work is not
+included in this project, but will be in a future FE project.
```

Comment:
Is there a jira ticket for this? If so, can we add it.

Reply:
Not yet.